cpp/src/slice2cs/Gen.cpp

```diff
@@ -1683,27 +1675,25 @@ Slice::Gen::TypesVisitor::visitInterfaceDefStart(const InterfaceDefPtr& p)
         baseNames.push_back("Ice.Object");
     }
 
-    //
-    // Check for cs:implements metadata.
-    //
-    const StringList metadata = p->getMetadata();
-    static const string prefix = "cs:implements:";
-    for (StringList::const_iterator q = metadata.begin(); q != metadata.end(); ++q)
+    // Check for `cs:implements` metadata.
+    for (const auto& metadata : p->getMetadata())
     {
-        if (q->find(prefix) == 0)
+        if (metadata->directive() == "cs:implements")
         {
-            baseNames.push_back(q->substr(prefix.size()));
+            baseNames.push_back(string(metadata->arguments()));
```

Comment:
Use `{}` for constructors

Reply:
I fixed all the ones I introduced in this PR.
Ripgripping around shows that's still many other places this should be fixed of course.